gcloud-java-dns/src/test/java/com/google/gcloud/dns/ZoneTest.java

```diff
@@ -0,0 +1,750 @@
+    assertNotNull(zone.dns());
+    assertEquals(dns, zone.dns());
+  }
+
+  @Test
+  public void testGetById() {
+    expect(dns.getZone(ZONE_ID)).andReturn(ZONE_INFO);
+    expect(dns.getZone(ZONE_ID, ZONE_FIELD_OPTIONS)).andReturn(ZONE_INFO); // for options
+    replay(dns);
+    Zone retrieved = Zone.get(dns, ZONE_ID);
+    assertSame(dns, retrieved.dns());
+    assertEquals(ZONE_INFO, retrieved.info());
+    BigInteger id = null;
+    try {
+      Zone.get(dns, id);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    try {
+      Zone.get(null, id);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    // test passing options
+    Zone.get(dns, ZONE_ID, ZONE_FIELD_OPTIONS);
+  }
+
+  @Test
+  public void testGetByName() {
+    expect(dns.getZone(ZONE_NAME)).andReturn(ZONE_INFO);
+    expect(dns.getZone(ZONE_ID, ZONE_FIELD_OPTIONS)).andReturn(ZONE_INFO); // for options
+    replay(dns);
+    Zone retrieved = Zone.get(dns, ZONE_NAME);
+    assertSame(dns, retrieved.dns());
+    assertEquals(ZONE_INFO, retrieved.info());
+    String name = null;
+    try {
+      Zone.get(dns, name);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    try {
+      Zone.get(null, name);
+      fail("Cannot get null zone.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    // test passing options
+    Zone.get(dns, ZONE_ID, ZONE_FIELD_OPTIONS);
+  }
+
+  @Test
+  public void deleteByIdAndFound() {
+    expect(dns.delete(ZONE_ID)).andReturn(true);
+    replay(dns);
+    boolean result = zone.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void deleteByIdAndNotFoundAndNameSetAndFound() {
+    expect(dns.delete(ZONE_ID)).andReturn(false);
+    expect(dns.delete(ZONE_NAME)).andReturn(true);
+    replay(dns);
+    boolean result = zone.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void deleteByIdAndNotFoundAndNameSetAndNotFound() {
+    expect(dns.delete(ZONE_ID)).andReturn(false);
+    expect(dns.delete(ZONE_NAME)).andReturn(false);
+    replay(dns);
+    boolean result = zone.delete();
+    assertFalse(result);
+  }
+
+  @Test
+  public void deleteByIdAndNotFoundAndNameNotSet() {
+    expect(dns.delete(ZONE_ID)).andReturn(false);
+    replay(dns);
+    boolean result = zoneNoName.delete();
+    assertFalse(result);
+  }
+
+  @Test
+  public void deleteByNameAndFound() {
+    expect(dns.delete(ZONE_NAME)).andReturn(true);
+    replay(dns);
+    boolean result = zoneNoId.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void deleteByNameAndNotFound() {
+    expect(dns.delete(ZONE_NAME)).andReturn(true);
+    replay(dns);
+    boolean result = zoneNoId.delete();
+    assertTrue(result);
+  }
+
+  @Test
+  public void listDnsRecordsByIdAndFound() {
+    Page<DnsRecord> pageMock = createStrictMock(Page.class);
```

Comment:
Chose (1) to fix this PR now and created issue #613. I should get to it later this week.